### DISCUSSION TITLE
Simple Feed Forward Network predicts better than DeepAR

### DISCUSSION THREAD

**Discussion by MarleneBs** [awslabs/gluonts#2430] [Q&A]
I have got different Data and the SImple Feed Forward Network predicts better than DeepAR. Is this possible ? The timeseries I predict are Partly also very volatile.
Please help me, I am very confused.

**Comment by mbohlkeschneider** [comment]
Hi @MarleneBs,

Thank you for your question. This can for sure happen, some models perform better on some datasets, other models on other ones. Overall, SimpleFeedForward is not too shabby (see this [paper](https://arxiv.org/pdf/2202.08485.pdf)).

**Reply by MarleneBs** [reply]
Thanks fpr your fast answer. Do you think it is possible that the Simple Feed Forward Estimator predicts datasets of electricity with an NRMSE of 0.024 or do you think I did something wrong? 
Would you mind to check this out for me?

**Reply by MarleneBs** [reply]
Where can I find the Paper? Maybe I have to cite it, so I need an official way to get it.

**Reply by mbohlkeschneider** [reply]
It is only on archive so far.

**Reply by MarleneBs** [reply]
So i actually cannot use it ? Thanks

**Reply by MarleneBs** [reply]
Could you maybe also check my Code? I am so afraid I have done something wrong, I need this for a Thesis and in some cases I got an NRMSE of 0.55 with DeepAR

**Reply by MarleneBs** [reply]
```
to_pandasdataset = lambda data: PandasDataset(
    data,
    target="value1",
    feat_dynamic_real=[feat1,feat2,feat3]
)
train = to_pandasdataset([ts.iloc[:-prediction_length, :] for ts in multiple_ts])
test = to_pandasdataset(multiple_ts)
```
In multiple_ts there is my data. 
(I gave the same pandasdataset to the Feed Forward, is this correct?)
```
trainer = Trainer(epochs=e, num_batches_per_epoch= batch_per_ep)
from gluonts.transform.sampler import ValidationSplitSampler
```
```
estimator_with_features = DeepAREstimator(
        freq= '60min',
        prediction_length= prediction_length,
        context_length= 2*prediction_length, #hat vielleicht ein bisschen was gebracht ?
        dropoutcell_type= 'RNNZoneoutCell', # hat vielleicht ein bisschen was gebracht?
        trainer= trainer,
        use_feat_dynamic_real= True,
        num_layers= nl,
        num_cells= nc,
        cell_type= 'lstm',
        #dropoutcell_type='ZoneoutCell',
        use_feat_static_cat= False ,
        use_feat_static_real= False,
        impute_missing_values= True, #was passiert dann?
        cardinality= None,
        batch_size= batch,
        validation_sampler= ValidationSplitSampler(),

    )
```
`predictor = estimator_with_features.train(train)`
```
forecast_it, ts_it = make_evaluation_predictions(dataset=test, predictor=predictor,num_samples= 100)
forecasts = list(forecast_it)
tests = list(ts_it)
evaluator = Evaluator(quantiles=[0.1, 0.5, 0.8, 0.95])
agg_metrics, item_metrics = evaluator(tests, forecasts, num_series=len(test))
```

**Reply by MarleneBs** [reply]
And for the Feed Forward I used the exact same data and:
```
from gluonts.model.simple_feedforward import SimpleFeedForwardEstimator
feedfoward = SimpleFeedForwardEstimator(
  prediction_length=prediction_length,
  context_length=prediction_length,
 trainer=Trainer(epochs=100, num_batches_per_epoch= batch_per_ep),
  batch_size= batch
)
predictor= feedfoward.train(train)
forecast_it, ts_it = make_evaluation_predictions(dataset=test, predictor=predictor)
forecasts = list(forecast_it)
tests = list(ts_it)
evaluator = Evaluator(quantiles=[0.1, 0.5, 0.8, 0.95])
agg_metrics, item_metrics = evaluator(tests, forecasts, num_series=len(test))
Is this correct?
```

**Reply by MarleneBs** [reply]
And with every run there is a very different NRMSE with DeepAR

**Reply by Saifbenhmida** [reply]
you need to fix the seed and you will find the same NRMSE for every run
